### PR TITLE
Vxr ssh admin prompt handling

### PR DESCRIPTION
**Pull request by shreyashah**
Spitfire devices both vxr and hardware types dont support admin capability.
But asr9k sim does support admin capability. However admin prompt in asr9k sim devices have an extra space character in the end of the prompt (after #) due to which config,commit,end-config() functionalities dont work as they have strict pattern matching which is pattern=r'\)#$'.
This now needs to be changed to pattern=r'\)#\s$'

Original techzone where this was witnessed:
https://techzone.cisco.com/t5/IOS-XR-PI-CQE-INFRA-Eng/PatternNotFoundException-Config-Prompt-not-found-after-sending/m-p/11757833

Testing logs:
allure.cisco.com/auto/cafy/release/latest/cafykit/work/archive/test_admin_dialog_20240411-152856_p4183355/reports/index.html

### CODE REVIEW

File: netmiko/cisco/cisco_vxr_ssh.py [hunk 1]
```diff
@@ -543,7 +547,10 @@ def commit(self, confirm=False, confirm_delay=None, comment='', label='', replac
                 msg = "Prompt not found after sending commit replace command"
                 self.log.error(msg)
                 raise PatternNotFoundException(msg)
-            expect_string = r'\)#$' + "|" + alt_error_marker
+            if self.base_prompt.startswith("sysadmin-vm:"):
```
Comment (akshansh237) [inline netmiko/cisco/cisco_vxr_ssh.py:550]: better to store in a boolean flag then checking this 3-4 times in the same method or store the expect string once and reuse same variable everywhere (append with alt_error_marker or other expected strings in special cases)

**Review by akshansh237 — CHANGES_REQUESTED** [review]
UT allure report has no logs. 
also add testcases for all the cases being handled here:
commit, commit replace, alt_error_marker, etc.

**Comment by shreyashah** [comment]
allure.cisco.com/auto/cafy/release/latest/cafykit/work/archive/test_admin_dialog_20240411-152856_p4183355/all.log shows the logs of admin config,commit

File: netmiko/cisco/cisco_vxr_ssh.py (hunk 1 above)
Comment (shreyashah) [reply netmiko/cisco/cisco_vxr_ssh.py:550]: Good idea. For now, I will leave it like this as the code logic works and has been tested with the below logs. 
The variable value deduplication will be tried to address in next PR.

**Pull request merged**